src/main/Main.java

```diff
@@ -0,0 +1,154 @@
+            ticketManager.generateTenTickets();
+            //testing getTicketById
+            System.out.println(ticketManager.getTicketById("ID1"));
+            //testing getTicketsByStadiumSector
+            List<Ticket> ticketsByStadiumSector = ticketManager.getTicketsByStadiumSector(Sector.A);
+            System.out.println(ticketsByStadiumSector.size());
+            //testing shared() by phone and by phone and email
+            emptyTicket.shared("123-456-789");
+            fullTicket.shared("123-456-789", "ticket@email.com");
+            //testing users polymorphism
+            AbstractUser client = new Client("Nice Client");
+            AbstractUser admin = new Admin("Good Admin");
+            client.printRole();
+            Ticket clientTicket = ((Client) client).getTicket();
+            admin.printRole();
+            boolean checkingResult = ((Admin) admin).checkTicket(clientTicket);
+            System.out.println("Checking result is: " + checkingResult);
+
+            //testing @NullableWarning
+            Ticket nullSectorFullTicket = new Ticket("ID02",
+                    "Concert",
+                    "001", eventTime,
+                    true,
+                    null,
+                    5.5);
+
+            //BusTicketValidatorTesting and reading from file
+            System.out.println("Read and validate busTickets: ");
+            BusTicketsValidator busTicketsValidator = new BusTicketsValidator();
+            String filePath = "src/main/files/busTickets.json";
+            List<BusTicket> busTickets = BusTicketReader.readTicketsFromFile(filePath);
+            busTicketsValidator.validateTickets(busTickets);
+*/
+            //ClientCoinArrayList testing
+
+            testExchangingCoinsForBonus();
+
+            // FineHashSet testing
+            testFineHashSet();
+
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private static void testFineHashSet() {
+        System.out.println("FineHashSet testing.....................");
+        FineHashSet fineSet = new FineHashSet();
+
+        Fine fine1 = new Fine(1, "Traveling without a ticket");
+        Fine fine2 = new Fine(2, "Traveling without a ticket for luggage");
+        Fine fine3 = new Fine(3, "Traveling without a ticket for an animal");
+        Fine fine4 = new Fine(4, "Drunken traveling");
+
+        fineSet.putFine(fine1);
+        fineSet.putFine(fine2);
+        fineSet.putFine(fine3);
+        fineSet.putFine(fine4);
+        System.out.println(fineSet.containsFine(2));
+        fineSet.deleteFine(3);
+        fineSet.iterateFines();
+    }
+
+    private static void testExchangingCoinsForBonus() {
+        System.out.println("ClientCoinArrayList testing.....................");
+        ClientCoinArrayList coinArrayList = new ClientCoinArrayList();
+
+        for (int i = 0; i < 10; i++) {
+            coinArrayList.putCoin(new Coin(CoinType.GOLD));
+        }
```

Comment:
can we use it just for each?

Reply:
We can't use a for-each loop because we are creating 10 new Coin objects and adding them to the collection. The for-each loop can only be used when we already have elements in the collection.